kqueue.go

```diff
@@ -71,10 +71,8 @@ func (w *Watcher) Close() error {
 		return nil
 	}
 	w.isClosed = true
-	w.mu.Unlock()
 
 	// copy paths to remove while locked
-	w.mu.Lock()
 	var pathsToRemove = make([]string, 0, len(w.watches))
 	for name := range w.watches {
```

Comment:
The loop can be simply replaced with `pathsToRemove = append(pathsToRemove, w.watches...)`.

Reply:
Make also ensures you make only one allocation for exactly what you need rather than continuously allocating on append (though append is smart and requests more than is needed to prevent multiple alloc requests)